functions.Rmd

```diff
@@ -14,6 +14,24 @@ row_adder()
 permutation()
 ```
 
+## Anonymous functions
+
+If using an anonymous or inline function (i.e. a function that hasn't been given a name by assigning it to a variable), prefer the new lambda syntax: `\(x) x + 1`.
+
+```R
+# Good
+map(xs, \(x) mean((x + 5)^2))
+
+# Bad
+map(xs, function(x) mean((x + 5)^2))
+map(xs, ~ mean((.x + 5)^2))
+```
+
+* As with any function, prefer informative argument names.
+* Only use an anonymous function if it fits on one line.
```

Comment:
Oh maybe this should have been expanded into an example - i think it is too subtle here

When I'm trying to decide between `\(x)` and `function(x)` I feel like I don't always start out knowing if its going to span >1 lines or not. So I tend to always reach for `function(x)` since that works for every case and I won't have to go back and switch `\ -> function` if I realize its a multiline function. That's part of why I think `function(x)` should be `# Ok` not `# Bad`